sender-sqs2/src/main/java/zipkin2/reporter/sqs2/SQSSender2.java

```diff
@@ -0,0 +1,171 @@
+
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import zipkin2.Call;
+import zipkin2.Callback;
+import zipkin2.codec.Encoding;
+import zipkin2.reporter.BytesMessageEncoder;
+import zipkin2.reporter.Sender;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+
+public class SQSSender2 extends Sender {
+
+  public static SQSSender2 create(String queueUrl) {
+    return newBuilder().sqsClient(SqsClient.create()).build();
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+
+    private SqsClient sqsClient;
+    private Encoding encoding = Encoding.JSON;
+    private String queueUrl;
+    private int messageMaxBytes = 256 * 1024; // 256KB SQS limit
+
+    public Builder queueUrl(String queueUrl) {
+      if (queueUrl == null) throw new NullPointerException("queueUrl == null");
+      this.queueUrl = queueUrl;
+      return this;
+    }
+
+    public Builder sqsClient(SqsClient sqsClient) {
+      if (sqsClient == null) throw new NullPointerException("sqsClient == null");
+      this.sqsClient = sqsClient;
+      return this;
+    }
+
+    public Builder messageMaxBytes(int messageMaxBytes) {
+      this.messageMaxBytes = messageMaxBytes;
+      return this;
+    }
+
+    public Builder encoding(Encoding encoding) {
+      this.encoding = encoding;
+      return this;
+    }
+
+    public SQSSender2 build() {
+      return new SQSSender2(this);
+    }
+
+    Builder(SQSSender2 sender) {
+      this.sqsClient = sender.sqsClient;
+      this.encoding = sender.encoding;
+      this.queueUrl = sender.queueUrl;
+      this.messageMaxBytes = sender.messageMaxBytes;
+    }
+
+    Builder() {}
+  }
+
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  private final SqsClient sqsClient;
+  private final Encoding encoding;
+  private final String queueUrl;
+  private final int messageMaxBytes;
+  volatile boolean closeCalled = false;
+
+  SQSSender2(Builder builder) {
+    this.sqsClient = builder.sqsClient;
+    this.encoding = builder.encoding;
+    this.queueUrl = builder.queueUrl;
+    this.messageMaxBytes = builder.messageMaxBytes;
+  }
+
+  @Override
+  public Encoding encoding() {
+    return this.encoding;
+  }
+
+  @Override
+  public int messageMaxBytes() {
+    return this.messageMaxBytes;
+  }
+
+  @Override
+  public int messageSizeInBytes(List<byte[]> list) {
+    int listSize = encoding().listSizeInBytes(list);
+    return (listSize + 2) * 4 / 3; // account for base64 encoding
+  }
+
+  static boolean isAscii(byte[] encodedSpans) {
+    for (int i = 0; i < encodedSpans.length; i++) {
+      if (encodedSpans[i] >= 0x80) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public final String toString() {
+    return "SQSSender2{queueUrl= " + queueUrl + "}";
```

Comment:
I would change the package name, but not the type name. Remember Kafka 0.10. wait.. 0.11.. crap 1.0.. aw man 2.3 :)